### DISCUSSION TITLE
Children props for customization inside the timer

### DISCUSSION THREAD

**Discussion by karimK3** [vydimitrov/react-countdown-circle-timer#102] [Q&A] [answered]
HI !

Is it possible to have an exemple of how to use children props to customize inside the timer ? 

I would like to format the timer with minutes and seconds but when I use the props children with your exemple it doesn't works.

const children = ({ remainingTime }) => {
  const minutes = Math.floor(remainingTime / 60)
  const seconds = remainingTime % 60

  return `${minutes}:${seconds}`
}

Nothing happen. I still have the timer with only seconds.

Thank you ! :)

**Comment by vydimitrov** [comment, marked as answer]
Hey, can I see how you use it?

**Reply by karimK3** [reply]
I use it like this but probably not the good way ...: 

```
<CountdownCircleTimer
                      isPlaying={true}
                      duration={100}
                      children={({ remainingTime }) => {
                        const minutes = Math.floor(remainingTime / 60)
                        const seconds = remainingTime % 60

                        return `${minutes}:${seconds}`
                      }}
                      strokeWidth={6}
                      size={WIDTH_SCREEN/4}
                      colors={[
                        ['#b9184f', 0.4],
                        ['#F7B801', 0.4],
                        ['#ff0057', 0.2],
                      ]}
                    >
```

**Reply by vydimitrov** [reply]
You should try this instead:

```js
<CountdownCircleTimer
  isPlaying
  duration={100}
  strokeWidth={6}
  size={WIDTH_SCREEN / 4}
  colors={[
    ['#b9184f', 0.4],
    ['#F7B801', 0.4],
    ['#ff0057', 0.2],
  ]}
>
  {({ remainingTime }) => {
    const minutes = Math.floor(remainingTime / 60);
    const seconds = remainingTime % 60;

    return `${minutes}:${seconds}`;
  }}
</CountdownCircleTimer>
```

**Reply by karimK3** [reply]
So yes, wasn't the right way. Now it works.
Thank you very much !

**Reply by karimK3** [reply]
Hi @vydimitrov :) 

When I use this code and I re-render my view after refreshing, remainingTime is always equal to zero. Do you know why ?

**Reply by vydimitrov** [reply]
Hey, you should change the `key` prop of the Timer to stat over

**Reply by Meishiiiiii** [reply]
How would you go about this now as when i try to do something similiar I get an error saying it needs to be within a text component? @vydimitrov

**Reply by vydimitrov** [reply]
May be just wrap it in a Text component?
